Userland/Libraries/LibSemVer/SemVer.cpp

```diff
@@ -0,0 +1,305 @@
+
+        // 1. Identifiers consisting of only digits are compared numerically
+        if (this_numerical_identifier.has_value() && other_numerical_identifier.has_value()) {
+            auto const this_value = this_numerical_identifier.value();
+            auto const other_value = other_numerical_identifier.value();
+
+            if (this_value == other_value) {
+                i++;
+                continue;
+            }
+
+            return this_value > other_value;
+        }
+
+        // 2. Identifiers with letters or hyphens are compared lexically in ASCII sort order
+        if (!this_numerical_identifier.has_value() && !other_numerical_identifier.has_value()) {
+            if (m_prerelease_identifiers[i] == other.m_prerelease_identifiers[i]) {
+                i++;
+                continue;
+            }
+            return m_prerelease_identifiers[i] > other.m_prerelease_identifiers[i];
+        }
+
+        // 3. Numeric identifiers always have lower precedence than non-numeric identifiers.
+        if (this_numerical_identifier.has_value() && !other_numerical_identifier.has_value())
+            return false;
+        if (!this_numerical_identifier.has_value() && other_numerical_identifier.has_value())
+            return true;
+    }
+
+    // 4. If all of the preceding identifiers are equal, larger set of pre-release fields has a higher precedence than a smaller set.
+    return m_prerelease_identifiers.size() > other.m_prerelease_identifiers.size();
+}
+
+bool SemVer::satisfies(StringView const& spec) const
+{
+    GenericLexer lexer(spec.trim_whitespace());
+    if (lexer.tell_remaining() == 0)
+        return false;
+
+    auto compare_op = lexer.consume_until<Function<bool(char)>>([](auto const& ch) {
+        return ch >= '0' && ch <= '9';
+    });
+
+    auto spec_version = MUST(from_string_view(lexer.consume_all()));
+    if (compare_op.length() == 0)
+        return is_same(spec_version, CompareType::Minor);
+    if (compare_op == "!=")
+        return !is_same(spec_version);
+
+    // Just match the version string
+    // Excluding prerelease and buildmetadata
+    if (compare_op == "="sv)
+        return is_same(spec_version, CompareType::Patch);
+
+    if (compare_op == "=="sv)
+        return is_same(spec_version);
+
+    // Current version is greater than spec
+    if (compare_op == ">"sv)
+        return is_greater_than(spec_version);
+    if (compare_op == "<"sv)
+        return is_lesser_than(spec_version);
+    if (compare_op == ">="sv)
+        return is_same(spec_version) || is_greater_than(spec_version);
+    if (compare_op == "<="sv)
+        return is_same(spec_version) || is_lesser_than(spec_version);
+
+    return false;
+}
+
+ErrorOr<SemVer> from_string_view(StringView const& version, char m_number_separator)
+{
+    using namespace regex;
+    if (version.count(m_number_separator) < 2)
+        return AK::Error::from_string_view("Insufficient version separator"sv);
```

Comment:
```suggestion
        return AK:Error::from_string_view("Insufficient occurrences of version separators"sv);
```

Just to make it clear what this is referring to.